scripts/go-release-automation.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+# Go Release Process Script
+# Used for copying go files to release directory and make a commit
+
+set -e  # Exit on error
+
+# Check if project name and version is provided
+if [ $# -ne 3 ]; then
+  echo "Usage: $0 <function> <project-name> [version]"
+  echo "Example: $0 get_release_dir_name ComAmazonawsKms v0.1.0"
+  echo "Example: $0 run_release_script ComAmazonawsKms v0.1.0"
+  exit 1
+fi
+
+# Function to map project name to release directory name
+get_release_dir_name() {
+  local project=$1
+  case "$project" in
+    "AwsCryptographicMaterialProviders") echo "mpl" ;;
+    "AwsCryptographyPrimitives") echo "primitives" ;;
+    "ComAmazonawsKms") echo "kms" ;;
+    "ComAmazonawsDynamodb") echo "dynamodb" ;;
+    "StandardLibrary") echo "smithy-dafny-standard-library" ;;
+    *) echo "Error: Unknown project name: $project" >&2; return 1 ;;
+  esac
+}
+
+run_release_script() {
+  PROJECT_NAME=$1
+  VERSION=$2
+
+  echo "Starting Go release script for $PROJECT_NAME $VERSION"
+
+  # Check if this is an AWS SDK project
+  IS_AWS_SDK=false
+  if [[ "$PROJECT_NAME" == "ComAmazonawsKms" || "$PROJECT_NAME" == "ComAmazonawsDynamodb" ]]; then
+    IS_AWS_SDK=true
+    echo "Detected AWS SDK project: $PROJECT_NAME"
+  fi
+
+  # Pull the latest smithy-dafny and libraries git submodules
+  echo " Pulling latest git submodules..."
+  git submodule update --init --recursive
```

Comment:
Some of these commands might have unintended consequences without user realizing it. Say if I develop on sub-module X.Y but then run the release, the release code might use different sub-module X.Z without me realizing it.

Reply:
I think we expect this to typically run in a GHA runner, so this seems fine